File: fn/src/main/java/org/sqlbuilder/fn/joins/Frame_CoreFEMember.java
package org.sqlbuilder.fn.joins;

import org.sqlbuilder.common.Insertable;

import java.util.HashSet;
import java.util.Set;

// TODO remove
public class Frame_CoreFEMember extends Pair<Integer, Integer> implements Insertable
{
	public static final Set<Frame_CoreFEMember> SET = new HashSet<>();

	// C O N S T R U C T O R

	public static Frame_CoreFEMember make(final int frameid, final int feid)
	{
		var fe = new Frame_CoreFEMember(frameid, feid);
		SET.add(fe);
		return fe;
	}

	private Frame_CoreFEMember(final int frameid, final int feid)
	{
		super(frameid, feid);
	}

	// I N S E R T

	@Override
	public String dataRow()
	{
		return String.format("%d,%d", first, second);
	}

	// T O S T R I N G

	@Override
	public String toString()
	{
		return String.format("[FR-coreFE frameid=%s feid=%s]", first, second);
	}
}

File: fn/src/main/java/org/sqlbuilder/fn/joins/Frame_CoreFEMember.kt
package org.sqlbuilder.fn.joins

import org.sqlbuilder.common.Insertable

// TODO remove
class Frame_CoreFEMember private constructor(
    frameid: Int,
    feid: Int,
) : Pair<Int, Int>(frameid, feid), Insertable {

    // I N S E R T

    override fun dataRow(): String {
        return "$first,$second"
    }

    // T O S T R I N G

    override fun toString(): String {
        return "[FR-coreFE frameid=$first feid=$second]"
    }

    companion object {

        val SET = HashSet<Frame_CoreFEMember>()

        fun make(frameid: Int, feid: Int): Frame_CoreFEMember {
            val fe = Frame_CoreFEMember(frameid, feid)
            SET.add(fe)
            return fe
        }
    }
}
